contracts/SnappAuction.sol

```diff
@@ -28,6 +28,14 @@ contract SnappAuction is SnappBase {
     uint public auctionIndex = MAX_UINT;
     mapping (uint => SnappBaseCore.PendingBatch) public auctions;
 
+    struct AuctionResult {
+        address solver;
+        uint objectiveValue;
+        bytes32 tentativeState;
+    }
+
+    mapping (uint => AuctionResult) public auctionResults;
```

Comment:
Can you comment what the key here is? Ideally this would be stored on the auction directly (assuming we refactor it to not use the PendingFlux struct).

Reply:
Yes. I was just about to address this `AuctionFlux` issue and noticed it would be absolutely necessary for the phase II anyway. However, I would personally prefer to merge this first and do the refactoring with the already existing `AuctionStruct`.

Reply:
Given that the current implementation needs fundamental changes in the way the phase checking is done, I'd prefer if we can only merge it once read. 

Are you worried about merge conflicts? I don't think you are touching any existing code here, so that should be fine.

Reply:
Yes, I can agree now that I saw the additional constraint regarding previous solution time.